t/05.runnabledb/java.t

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env perl
+
+# Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+# Copyright [2016-2019] EMBL-European Bioinformatics Institute
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#      http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+use strict;
+use warnings;
+
+use Cwd            ();
+use File::Basename ();
+$ENV{'EHIVE_ROOT_DIR'} ||= File::Basename::dirname( File::Basename::dirname( File::Basename::dirname( Cwd::realpath($0) ) ) );
```

Comment:
This should be //= rather than ||=, even though it isn't exactly very likely for someone to set EHIVE_ROOT_DIR to '0'.

Reply:
This exists in most of the test plans under t/, so it should probably be fixed across all of them in a separate PR. As for the whitespace issues --  although I also prefer whitespace around the `.` operator and like to see it after if/unless, what's done is consistent with existing coding style. Also, for consistency, I'm reluctant to make changes in between-version merge-ups unless necessary to avoid a conflict.

I propose that these changes be brought into master from 2.5 as is, we keep the whitespace issue in mind and endeavour to fix when the opportunity arises in future PRs, and fix the `||=` vs `//=` issue across the repo in a future refactoring.

Reply:
Fair enough.